### PR TITLE
Add all recipes of the Humanoid Path Planner (HPP) suite

### PR DESCRIPTION
**Pull request by ymontmarin**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by ymontmarin** [comment]
@chrisburr Thank you very much for your help ! The dependency should be OK for osx to finish compiling ;)

But I think the issues I got with OpenGL and X11 on those recipes and other ones are more general. I have few questions.

- What is the best strategy between using cdt and xorg-* package for linux recipes ? The recipes I build strongly rely on qt which only use cdt
- Is there any equivalent of cdt for osx ? I know that OpenGL is naturally found in Apple framework but what about X11 which come with Xquartz ?
- Furthermore, I have some compatibility issue due to some package using cocoa/OpenGL vs some using X11/OpenGL

Any insights on those questions would be welcome !
Best

File: recipes/hpp-gui/meta.yaml
Comment (chrisburr) [inline recipes/hpp-gui/meta.yaml:36]: Our `boost` package is `boost-cpp` + Boost.Python. If not, using Boost.Python you can remove `boost` from `host` and `run`.

File: recipes/hpp-gui/meta.yaml
Comment (chrisburr) [inline recipes/hpp-gui/meta.yaml:57]: See above

File: recipes/hpp-gui/meta.yaml
Comment (chrisburr) [inline recipes/hpp-gui/meta.yaml:77]: This should be wrapped:
```suggestion
    HPP (Humanoid Path Planner) is a collection of software packages implementing
    path planning functionalities for a humanoid robot. The core functionalities are well
    suited to solve classical path planning problems. It can be extended to path planning
    for other types of robots like digital actors for instance.
```

File: recipes/hpp-gui/COPYING
Comment (chrisburr) [inline recipes/hpp-gui/COPYING:1]: You only need to include the license files if they're not included in the source tarballs, as you're using Github releases they should be present.

File: recipes/hpp-environments/build.sh
Comment (chrisburr) [inline recipes/hpp-environments/build.sh:9]: Do you need this line? It should be in conda-forge's default C++ flags:
```suggestion
```

File: recipes/hpp-corbaserver/meta.yaml
Comment (chrisburr) [inline recipes/hpp-corbaserver/meta.yaml:54]:
```suggestion
    - test -f $PREFIX/lib/lib{{ name }}${SHLIB_EXT}  # [unix]
```

**Review by chrisburr — COMMENTED** [review]
These comments apply to the other recipes as well

**Comment by chrisburr** [comment]
Unlike Anaconda Inc, conda-forge pretty much only uses CDTs for OpenGL on Linux, see [here for details on how to do it correctly](https://conda-forge.org/docs/maintainer/knowledge_base.html#libgl). Everything else should be coming from conda-forge directly.

For macOS you might have more luck disabling it for now and re-adding it in a pull request to the feedstock (feel free to ping me on that PR for a review if you have questions). At the very least, the CI will be much faster as there will be one CI job per Python version instead of chaining them together. You might also find you need to use a newer macOS SDK, which isn't possible in staged-recipes (see [here](https://conda-forge.org/docs/maintainer/knowledge_base.html#requiring-newer-macos-sdks)). You should be able to find all of the qt/x11 packages you need for conda-forge directly so only OpenGL will be taken from the macOS SDK. If not, we can take it from there.

File: recipes/hpp-gui/meta.yaml
Comment (ymontmarin) [reply recipes/hpp-gui/meta.yaml:36]: Thanks for the precision, will revise that

File: recipes/hpp-gui/meta.yaml
Comment (ymontmarin) [reply recipes/hpp-gui/meta.yaml:77]: Will do !

File: recipes/hpp-gui/COPYING
Comment (ymontmarin) [reply recipes/hpp-gui/COPYING:1]: The repo of some of the recipe does not contain the file, some does. As they all point to the same organisation and global project. Is this approach ok with conda-forge or should we wait our next github release that will include file ?

File: recipes/hpp-environments/build.sh
Comment (ymontmarin) [reply recipes/hpp-environments/build.sh:9]: The default value is c++14 for osx and i  that helped me to fix some osx issue to force 17 standard, is it ok ?

File: recipes/hpp-corbaserver/meta.yaml
Comment (ymontmarin) [reply recipes/hpp-corbaserver/meta.yaml:54]: Thank you for your tips !

**Comment by ymontmarin** [comment]
@chrisburr Thank you for your insights, I follow those advice to clean other recipes I maintain and have cleaner dependency / cdt use.

When I am finish with those recipes I can finish this PR with the last fix and the correction you suggest !

File: recipes/hpp-gui/COPYING
Comment (chrisburr) [reply recipes/hpp-gui/COPYING:1]: Where do the license files come from? This way is fine provided the files are "approved" by the upstream copyright holders.

Comment (ymontmarin) [reply recipes/hpp-gui/COPYING:1]: This is a common fille present in some of the package but not on some others. This is the common file on the project, I only patch the package that does not have one. But I will confirm with upstream before merge !

**Comment by ymontmarin** [comment]
@chrisburr I think this is now ready to merge ! Thank you very much for your help

**Review by chrisburr — CHANGES_REQUESTED** [review]
Following up to https://github.com/conda-forge/staged-recipes/pull/13033#discussion_r516293545, did you confirm that upstream agrees with these licenses being applied? It would be good to make pull requests to add the license files and get them accepted to know for sure.

It's also important that they're happy to have the license applied retro-actively to older versions as I've worked with packages where this wasn't the case.

**Comment by ymontmarin** [comment]
@chrisburr Yes of course, I will validate with upstream first ! But the recipe code are ready. I'll get back to you when the license point is clarified

**Comment by jmirabel** [comment]
Dear @chrisburr and @ymontmarin,

I am one of the main developers of HPP and I confirm all the packages of the [HPP organization](https://github.com/humanoid-path-planner/) are released under LGPL license, except otherwise mentioned by a license file in each specific package. This applies from the creation of each package (whenever there is no license file in the package).

All the files (except maybe rare examples) contains a license header which confirms the above.

Thank you both for your work.

**Review by chrisburr — APPROVED** [review]

**Comment by chrisburr** [comment]
Thank you very much @jmirabel for passing by to confirm!

LGTM! 👍

**Comment by ymontmarin** [comment]
Thank you @chrisburr  !

**Pull request merged**
